core/src/main/java/nl/nn/adapterframework/errormessageformatters/XslErrorMessageFormatter.java

```diff
@@ -60,45 +64,40 @@ public Message format(String errorMessage, Throwable t, INamedObject location, M
 				Transformer errorTransformer;
 
 				if (StringUtils.isNotEmpty(getStyleSheet())) {
-					URL url = ClassUtils.getResourceURL(this, styleSheet);
-					errorTransformer = XmlUtils.createTransformer(url);
-				}
-				else {
-					String xpath = getXpathExpression();
-//					if (StringUtils.isEmpty(xpath)) {
-//						xpath="/errorMessage/@message";
-//					}
-					errorTransformer = XmlUtils.createTransformer(XmlUtils.createXPathEvaluatorSource(xpath));
+					URL url = ClassUtils.getResourceURL(this, getStyleSheet());
+					int xsltVersion = XmlUtils.detectXsltVersion(url);
+					errorTransformer = XmlUtils.createTransformer(new StreamSource(url.openStream()), xsltVersion, new ClassLoaderURIResolver(this));
```

Comment:
It might be better to use a Resource here. It will also set the systemId, reads from the url (with the correct resolver). The only thing I'm not sure about is if the transformerpool uses the SaxSource resourceResolver